### PR TITLE
Cortx 32236 te rvf multi part with versioning scenarios part2

### PR DESCRIPTION
**Pull request by DhobaleVishal**
# Problem Statement
- TE MultiPart Versioning scenario part2.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
    TP: TEST-43544
    [TEST-41288_TestVersioningMultipart_20220616022202.log](https://jts.seagate.com/secure/attachment/522453/TEST-41288_TestVersioningMultipart_20220616022202.log)
    [TEST-41289_TestVersioningMultipart_20220616024016.log](https://jts.seagate.com/secure/attachment/522454/TEST-41289_TestVersioningMultipart_20220616024016.log)
    [TEST-41290_TestVersioningMultipart_20220616034443.log](https://jts.seagate.com/secure/attachment/522455/TEST-41290_TestVersioningMultipart_20220616034443.log)
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [x] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: libs/s3/s3_multipart.py [hunk 1]
```diff
@@ -317,10 +276,7 @@ def multipart_download(self, **kwargs):
...
         try:
             self.s3_resource.Object(bucket_name, key). \
-                download_file(file_path,
-                              Config=config,
-                              Callback=ProgressPercentage(file_path)
-                              )
+                download_file(file_path, Config=config, Callback=ProgressPercentage(file_path))
         except Exception as error:
-            LOGGER.error("Error in multipart_upload: %s", str(error))
+            LOGGER.exception("Error in multipart_upload: %s", str(error))
```
Comment (apoorva-rao) [inline libs/s3/s3_multipart.py:281]: Let's update this to "Error in multipart_download" and while we're at it, we can also use the newly introduced constant - ERR_MSG

File: libs/s3/s3_multipart_test_lib.py [hunk 2]
```diff
@@ -175,35 +153,24 @@ def upload_parts(self,
...
-                                   **kwargs) -> tuple:
+    def upload_precalculated_parts(self, upload_id: int = None, bucket_name: str = None,
+                                   object_name: str = None, **kwargs) -> tuple:
         """
-        Upload specified/precalculated part sizes for a specific multipart upload ID one part
-        at a time.
+        Upload precalculated parts.
 
+        Upload specified/precalculated part sizes for a specific multipart uploadID one part at
+        a time.
```
Comment (apoorva-rao) [inline libs/s3/s3_multipart_test_lib.py:173]: Let's keep either line 170 or 172-173. Repetition in the docstring seems unnecessary

**Review by apoorva-rao — CHANGES_REQUESTED** [review]
@DhobaleVishal minor comments added inline

**Review by apoorva-rao — APPROVED** [review]

**Comment by archanabpatil** [comment]
test_mpu_ver_enabled_bkt_41285
test_preexist_mpu_versioning_enabled_bkt_41284
test_mpu_versioning_suspended_bkt_41286
test_mpu_del_versioning_enabled_bkt_41287

are these tests also ran ? Because I see the code is merged into single file as we dicided already. Just need to recheck once if nothing in these 4 tests is broken due to the merge.  First 3 tests were passing and last was failing . So please make sure the result is consistent @DhobaleVishal

**Comment by DhobaleVishal** [comment]
> test_mpu_ver_enabled_bkt_41285 test_preexist_mpu_versioning_enabled_bkt_41284 test_mpu_versioning_suspended_bkt_41286 test_mpu_del_versioning_enabled_bkt_41287
> 
> are these tests also ran ? Because I see the code is merged into single file as we dicided already. Just need to recheck once if nothing in these 4 tests is broken due to the merge. First 3 tests were passing and last was failing . So please make sure the result is consistent @DhobaleVishal

@archanabpatil Yes, result is consistent.
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_versioning_test_lib.py: 175]: Successfully deleted the object's version: {'ResponseMetadata': {'HTTPStatusCode': 204, 'HTTPHeaders': {'x-amz-version-id': 'zWqhrHGimgR7F1TTkqb5ChCouFBjfh0', 'date': 'Mon, 20 Jun 2022 10:14:56 GMT', 'connection': 'Keep-Alive'}, 'RetryAttempts': 0}, 'VersionId': 'zWqhrHGimgR7F1TTkqb5ChCouFBjfh0'}
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_test_lib.py: 365]: You have opted to delete buckets.
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_test_lib.py: 368]: Trying polling mechanism as bucket is getting deleted forcefully.
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_test_lib.py: 228]: Listing Objects from bucket: s3bkt-versioning-3985597242202595
[2022-06-20 04:14:56] [MainThread] [DEBUG ] [s3_core_lib.py: 214]: []
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_core_lib.py: 324]: This might cause data loss as you have opted for bucket deletion with objects in it
[2022-06-20 04:14:56] [MainThread] [DEBUG ] [s3_core_lib.py: 328]: Objects deleted successfully from bucket s3bkt-versioning-3985597242202595, response: []
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_test_lib.py: 228]: Listing Objects from bucket: s3bkt-versioning-3985597242202595
[2022-06-20 04:14:56] [MainThread] [DEBUG ] [s3_core_lib.py: 214]: []
[2022-06-20 04:14:56] [MainThread] [DEBUG ] [s3_core_lib.py: 331]: Bucket 's3bkt-versioning-3985597242202595' deleted successfully. Response: {'ResponseMetadata': {'HTTPStatusCode': 204, 'HTTPHeaders': {'date': 'Mon, 20 Jun 2022 10:14:56 GMT', 'connection': 'Keep-Alive'}, 'RetryAttempts': 0}}
[2022-06-20 04:14:56] [MainThread] [DEBUG ] [s3_test_lib.py: 373]: {'ResponseMetadata': {'HTTPStatusCode': 204, 'HTTPHeaders': {'date': 'Mon, 20 Jun 2022 10:14:56 GMT', 'connection': 'Keep-Alive'}, 'RetryAttempts': 0}}
[2022-06-20 04:14:56] [MainThread] [INFO  ] [s3_test_lib.py: 376]: ############# BUCKET DELETION TIME : 0.389881 #############
[2022-06-20 04:14:56] [MainThread] [INFO  ] [test_versioning_multipart.py: 80]: ENDED: Teardown operations.
============================================================================ short test summary info ============================================================================
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_preexist_mpu_versioning_enabled_bkt_41284
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_mpu_ver_enabled_bkt_41285
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_mpu_versioning_suspended_bkt_41286
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_abort_multipart_upload_does_not_create_a_new_version_41288
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_upload_multiple_versions_to_multipart_uploaded_object_in_versioned_bucket_41289
PASSED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_upload_new_versions_to_existing_objects_using_multipart_upload_41290
FAILED tests/s3/versioning/test_versioning_multipart.py::TestVersioningMultipart::test_mpu_del_versioning_enabled_bkt_41287 - TypeError: delete_version() missing 1 required p...
============================================================== 1 failed, 6 passed, 12 warnings in 97.00s (0:01:36) ==============================================================
(virenv) [root@ssc-vm-g3-rhev4-3019 cortx-test]#

**Review by archanabpatil — APPROVED** [review]

**Review by VikasKumarC — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
Approving for merge!!!!

**Pull request merged**